cli/running/watchdog.go

```diff
@@ -73,16 +71,11 @@ func (wd *Watchdog) Start() {
 		}
 		wd.logger = wd.Instance.logger
 		// Start the Instance and forwarding signals (except  SIGINT and SIGTERM)
-		wd.startSignalHandling()
-		wd.stopMutex.Lock()
-		if !wd.stopped {
-			if err := wd.Instance.Start(); err != nil {
-				wd.logger.Printf(`Watchdog(ERROR): "%v".`, err)
-				wd.stopMutex.Unlock()
-				break
-			}
+		if err := wd.Instance.Start(); err != nil {
+			wd.logger.Printf(`Watchdog(ERROR): "%v".`, err)
+			break
 		}
-		wd.stopMutex.Unlock()
+		wd.startSignalHandling()
```

Comment:
Looks like, we have a gap between Tarantool start and signal handling start. If I'm not mistaken, SIGINT or SIGTERM received here will not be handled correctly and tt process will be terminated. But Tarantool will continue execution. Please, check.

Reply:
No, we have registered a handler for those signals before it and have not unregistered. Yes, the handler will not work in this function, but the process will not be terminated. The effect will be the same as by ignoring the signal.

Reply:
I have replaced using contexts with channels and added an ignoring signals before the start of watchdog, please check it.

Reply:
Will be SIGTERM & SIGINT silently ignored in this case?

Reply:
Yes, until we start a watchdog's handling loop it will be ignored.

Reply:
> Yes, until we start a watchdog's handling loop it will be ignored.

So from user's point of view it looks like non-working `tt stop` :) This is not so critical as the start-stop race, but it's still an issue.